actionpack/lib/abstract_controller/rendering.rb

```diff
@@ -123,10 +123,6 @@ def _set_rendered_content_type(format) # :nodoc:
     # :api: private
     def _normalize_render(*args, &block)
       options = _normalize_args(*args, &block)
-      #TODO: remove defined? when we restore AP <=> AV dependency
-      if defined?(request) && request.variant.present?
-        options[:variant] = request.variant
```

Comment:
An option is to extract a method and add it to the public API so concrete implementations can implement this method. Action Controller would implement it grabbing from the request, Action Mailer would use the default implementation (do nothing) and other subclasses could just implement this same method.